binary_search_tree/tree.py

```diff
@@ -14,21 +14,106 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n) if the tree is balanced
+    # Space Complexity: O(1) - we are just creating one object (the new node)
+    
+    def add_helper(self, current_root, new_node):
+        #we can treat each sub-tree like it's a new tree
+        if current_root is None:
+            return new_node
+        
+        if new_node.key < current_root.key:
+            current_root.left = self.add_helper(current_root.left, new_node)
+        else:
+            current_root.right = self.add_helper(current_root.right, new_node)
+
+        return current_root
+
+
     def add(self, key, value = None):
-        pass
+        new_node = TreeNode(key, value) #creating the node we are going to add
+        if self.root == None:
+            self.root = new_node
+            return self.root 
+
+        current = self.root 
+        while current:
+            if current.key > key: 
+                #go left
+                if current.left is None: #I am at the bottom
+                    #so add the node
+                    current.left = new_node
+                else:
+                    current = current.left
+            else:
+                #go right
+                if current.right is None:
+                    current.right = new_node
+                else:
+                    current = current.right
+        #Since there is a root,
+        #Checking whether the new value is bigger than the root
+        else:
+            if value < self.root.value:
+                self.root.left = self.add(self.root.left, value)
+            else:
+                self.root.right = self.add(self.root.right, value)
+
+            return self.root
+
 
     # Time Complexity: 
     # Space Complexity: 
     def find(self, key):
-        pass
+        #check if there's anything in the tree
+        if self.root == None:
+            return None
+        
+        if self.root.key == key:
+            return self.root.value
+        
+        current = self.root
+        while current:
+            if current.key > key:
```

Comment:
You're on the right track here. Since it's a binary search tree, if you the key of the current node we are looking at is greater than the key we are searching for, we want to search the left subtree. Otherwise, we want to search the right subtree. 

Each time we iterate through our while loop again and are iterating over a new node in the tree, we want to check if the current node's key is the key we are looking for. If it is, you can return that node's value.

Otherwise, keep using the key to check if you want to go down the left or right subtree. If you reach a leaf and still haven't found a match, you can return `None` because you know there is no node in the tree with that key. 

It follows a similar pattern to `add` above. It just has a twist